sdk/storage/storage-queue/test/utils/recorder.ts

```diff
@@ -0,0 +1,266 @@
+      recordings = (window as any).__json__["recordings/" + fp].recordings;
+      uniqueTestInfo = (window as any).__json__["recordings/" + fp].uniqueTestInfo;
+
+      xhr.onCreate = function(req: any) {
+        const reqSend = req.send;
+        req.send = async function(data: any) {
+          reqSend.call(req, data);
+
+          const parsedUrl = queryString.parseUrl(req.url);
+          let formattedRequest = {
+            method: req.method,
+            url: parsedUrl.url,
+            query: parsedUrl.query,
+            requestBody: (data instanceof Blob) ? await blobToString(data) : data
+          };
+
+          let recordingFound = false;
+          for (let i = 0; !recordingFound && i < recordings.length; i++) {
+            if (matchRequest(recordings[i], formattedRequest)) {
+              let status = recordings[i].status;
+              let responseHeaders = recordings[i].responseHeaders;
+              let response = recordings[i].response;
+              setTimeout(() => req.respond(status, responseHeaders, response));
+              recordings.splice(i, 1);
+              recordingFound = true;
+            }
+          }
+
+          if (!recordingFound) {
+            throw new Error("No match for request " + JSON.stringify(formattedRequest, null, " "));
+          }
+        }
+      }
+
+      // We're not matching request headers
+      function matchRequest(recording: any, request: any) {
+        for (let param in recording.query) {
+          if (recording.query[param] !== request.query[param] && param !== "_") {
+            return false;
+          }
+        }
+
+        for (let param in request.query) {
+          if (recording.query[param] === undefined) {
+            return false;
+          }
+        }
+
+        return (
+          recording.method === request.method &&
+          recording.url === request.url &&
+          recording.requestBody === request.requestBody
+        );
+      }
+    },
+    stop: function() {
+      console.log(JSON.stringify({
+        writeFile: true,
+        path: "./recordings/" + fp,
+        content: { recordings: recordings, uniqueTestInfo: uniqueTestInfo }
+      }));
+      xhr.restore();
+    }
+  };
+}
+
+export function record(this: any, folderpath: string, testTitle?: string) {
+  let recorder: any;
+  let isRecording: boolean;
+  let isPlayingBack: boolean;
+
+  if (isBrowser()) {
+    recorder = niseRecorder(folderpath, testTitle || this.currentTest.title);
+    isRecording = ((window as any).__env__.TEST_MODE === "record");
+    isPlayingBack = ((window as any).__env__.TEST_MODE === "playback");
+  } else {
+    recorder = nockRecorder(folderpath, testTitle || this.currentTest.title);
+    isRecording = (process.env.TEST_MODE === "record");
+    isPlayingBack = (process.env.TEST_MODE === "playback");
+  }
+
+  if (recorder.skip && (isRecording || isPlayingBack)) {
+    this.skip();
+  }
+
+  if (isRecording) {
```

Comment:
I think we have decided on setting TEST_MODE = "record" by default ?

Reply:
We have decided to keep the original behavior by default. You have updated it in the description. It's a good idea to be able to skip recording/playback, otherwise some tests would always be skipped.

Reply:
Oh yeah, it has been so long :)